### PR TITLE
[CIVIS] Revamp setup instructions for Gitlab

### PR DESCRIPTION
**Pull request by AdrianLC**
### What does this PR do? What is the motivation?
Updated to Gitlab Visibility docs:
- Fixes outdated instructions for Gitlab logs collection.
- Adds some disclaimer notes about support in infrastructure metrics correlation and logs collection.
- Moves the webhook setup instructions under a tab for Gitlab < 13.7

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

**Review by alxch- — COMMENTED** [review]
LGTM from eng standpoint

File: content/en/continuous_integration/pipelines/gitlab.md [hunk 1]
```diff
@@ -97,6 +97,30 @@ Then, configure the integration on a [project][3] by going to **Settings > Integ
 [2]: https://docs.gitlab.com/ee/administration/operations/rails_console.html#using-the-rails-runner
 [3]: https://docs.gitlab.com/ee/user/admin_area/settings/project_integration_management.html#use-custom-settings-for-a-group-or-project-integration
 {{% /tab %}}
+
+{{% tab "GitLab &lt; 13.7" %}}
+
+For versions older than when the native Datadog integration, you can use [webhooks][1] to send pipeline data to Datadog.
```
Comment (cswatt) [inline content/en/continuous_integration/pipelines/gitlab.md:103]:
```suggestion
For older versions of GitLab, you can use [webhooks][1] to send pipeline data to Datadog.
```

File: content/en/continuous_integration/pipelines/gitlab.md [hunk 2]
```diff
@@ -97,6 +97,30 @@ Then, configure the integration on a [project][3] by going to **Settings > Integ
...
+
+<div class="alert alert-info"><strong>Note</strong>: Support directly with webhooks is not actively under development. Unexpected issues could happen. Updating the GitLab version is recommended instead.</div>
```
Comment (cswatt) [inline content/en/continuous_integration/pipelines/gitlab.md:105]:
```suggestion
<div class="alert alert-info"><strong>Note</strong>: Direct support with webhooks is not under development. Unexpected issues could happen. Datadog recommends that you update GitLab instead.</div>
```

File: content/en/continuous_integration/pipelines/gitlab.md [hunk 3]
```diff
@@ -240,18 +248,42 @@ See the table below for the message and domain correlated with each error type.
...
+
+For more information about processing job logs collected from the GitLab integration, see the [Processors documentation][17].
 
 To enable collection of job logs:
 
-1. Enable the `datadog_integration_logs_collection` [feature flag][8] in your GitLab self-hosted or GitLab.com account. This allows you to see the **Enable job logs collection** checkbox on the [Pipeline Setup page][17].
-2. Click **Enable job logs collection** and click **Save changes**.
+{{< tabs >}}
+{{% tab "GitLab.com" %}}
+1. Click **Enable job logs collection** checkbox in the GitLab integration **Settings > Integrations > Datadog**.
```
Comment (cswatt) [inline content/en/continuous_integration/pipelines/gitlab.md:264]:
```suggestion
1. Click the **Enable job logs collection** checkbox in the GitLab integration **Settings > Integrations > Datadog**.
```

File: content/en/continuous_integration/pipelines/gitlab.md [hunk 4]
```diff
@@ -240,18 +248,42 @@ See the table below for the message and domain correlated with each error type.
...
+2. Click **Save changes**.
+{{% /tab %}}
 
-Job logs are collected in [Log Management][9] and are automatically correlated with the GitLab pipeline in CI Visibility. Log files larger than one GiB are truncated.
+{{% tab "GitLab &gt;&equals; 15.3" %}}
+<div class="alert alert-info"><strong>Note</strong>: Datadog downloads log files directly from your GitLab logs [object storage][1] with temporary pre-signed URLs.
+The storage must not have network restrictions such as IP range whitelist.</div>
```
Comment (cswatt) [inline content/en/continuous_integration/pipelines/gitlab.md:270]:
```suggestion
The storage must not have network restrictions, such as an IP range allowlist.</div>
```

File: content/en/continuous_integration/pipelines/gitlab.md [hunk 5]
```diff
@@ -240,18 +248,42 @@ See the table below for the message and domain correlated with each error type.
...
-2. Click **Enable job logs collection** and click **Save changes**.
+{{< tabs >}}
+{{% tab "GitLab.com" %}}
+1. Click **Enable job logs collection** checkbox in the GitLab integration **Settings > Integrations > Datadog**.
+2. Click **Save changes**.
+{{% /tab %}}
 
-Job logs are collected in [Log Management][9] and are automatically correlated with the GitLab pipeline in CI Visibility. Log files larger than one GiB are truncated.
+{{% tab "GitLab &gt;&equals; 15.3" %}}
+<div class="alert alert-info"><strong>Note</strong>: Datadog downloads log files directly from your GitLab logs [object storage][1] with temporary pre-signed URLs.
```
Comment (cswatt) [inline content/en/continuous_integration/pipelines/gitlab.md:269]:
```suggestion
<div class="alert alert-info"><strong>Note</strong>: Datadog downloads log files directly from your GitLab logs <a href="https://docs.gitlab.com/ee/administration/job_artifacts.html#using-object-storage">object storage</a> with temporary pre-signed URLs.
```

File: content/en/continuous_integration/pipelines/gitlab.md [hunk 6]
```diff
@@ -240,18 +248,42 @@ See the table below for the message and domain correlated with each error type.
...
+The storage must not have network restrictions such as IP range whitelist.</div>
 
-<div class="alert alert-info"><strong>Note</strong>: Logs are billed separately from CI Visibility.</div>
+1. Click **Enable job logs collection** checkbox in the GitLab integration **Settings > Integrations > Datadog**.
+2. Click **Save changes**.
+
+[1]: https://docs.gitlab.com/ee/administration/job_artifacts.html#using-object-storage
```
Comment (cswatt) [inline content/en/continuous_integration/pipelines/gitlab.md:275]:
```suggestion
```

File: content/en/continuous_integration/pipelines/gitlab.md [hunk 7]
```diff
@@ -240,18 +248,42 @@ See the table below for the message and domain correlated with each error type.
...
+2. Click **Save changes**.
+{{% /tab %}}
 
-Job logs are collected in [Log Management][9] and are automatically correlated with the GitLab pipeline in CI Visibility. Log files larger than one GiB are truncated.
+{{% tab "GitLab &gt;&equals; 15.3" %}}
+<div class="alert alert-info"><strong>Note</strong>: Datadog downloads log files directly from your GitLab logs [object storage][1] with temporary pre-signed URLs.
+The storage must not have network restrictions such as IP range whitelist.</div>
 
-<div class="alert alert-info"><strong>Note</strong>: Logs are billed separately from CI Visibility.</div>
+1. Click **Enable job logs collection** checkbox in the GitLab integration **Settings > Integrations > Datadog**.
```
Comment (cswatt) [inline content/en/continuous_integration/pipelines/gitlab.md:272]:
```suggestion
1. Click **Enable job logs collection** checkbox in the GitLab integration under **Settings > Integrations > Datadog**.
```

File: content/en/continuous_integration/pipelines/gitlab.md [hunk 8]
```diff
@@ -240,18 +248,42 @@ See the table below for the message and domain correlated with each error type.
...
+2. Click **Save changes**.
+
+[1]: https://docs.gitlab.com/ee/administration/job_artifacts.html#using-object-storage
+{{% /tab %}}
+
+{{% tab "GitLab &gt;&equals; 14.8" %}}
+1. Enable the `datadog_integration_logs_collection` [feature flag][1] in your GitLab. This allows you to see the **Enable job logs collection** checkbox in the GitLab integration **Settings > Integrations > Datadog**.
```
Comment (cswatt) [inline content/en/continuous_integration/pipelines/gitlab.md:279]:
```suggestion
1. Enable the `datadog_integration_logs_collection` [feature flag][1] in your GitLab. This allows you to see the **Enable job logs collection** checkbox in the GitLab integration under **Settings > Integrations > Datadog**.
```

**Review by cswatt — APPROVED** [review]

**Pull request merged**